### DISCUSSION TITLE
Speedrun Support

### DISCUSSION THREAD

**Discussion by DanielBatesJ** [StrataSource/Portal-2-Community-Edition#515] [Ideas]
The speedrunning community is excited to hopefully at some point in the future either also run, or primarily run P2CE. I'll try to use this discussion page for requests and information for the team when they have the time to look at some of our suggestions.

1. **Plugin Functionality**: The plugin we use currently ([Source Auto Record or SAR](https://github.com/Blenderiste09/SourceAutoRecord)) adds a massive amount of Quality of Life, as well as making timing and proof more accurate. The SAR team is very active, and all of the Source Code for the project is public. A good number of the features we have are listed on our [wiki](https://wiki.portal2.sr/SAR), but there are many more, most of which are used often by the community.
2. **Leaderboards**:  The [current leaderboards](https://board.iverb.me/changelog) pull from the public steam leaderboards API. But the information is limited. The site allows us to moderate the leaderboards (ban times/players), as well as allows users to upload proof. The hope with the new leaderboards that are currently under development is to allow for support of custom categories for maps (think custom routes or even the custom chamber variations from Portal 1), and to bundle cooperative times between two players  (not something that the Steam leaderboards supports, at least with Portal 2). Things like automatic demo uploading and coop times being bundled together would be amazing tools for the development team, assuming speedrunners do move over to P2CE.

Feel free to ask us any follow up questions, the leaderboard development team as well as the SAR development team is more than open to working with you guys to the extend you need.

**Comment by JJL772** [comment]
1. Since we're partnered with momentum, we can probably use some of their features to replace SAR. I'll discuss this with the team more in the future.

2. We can look into implementing many of the leaderboard suggestions. Bundling player score with co-op might be a bit tricky, as I don't know that much about Steam leaderboards, but I will look into it more. Other features would be pretty trivial to support though. 

Thanks for reaching out to us, we really want to collaborate with the speedrunning community on stuff like this!

**Reply by JJL772** [reply]
To follow up on this: @mlugg and @Blenderiste09 are now apart of the team and will be handling the implementation of speedrun features.
